File: z-clients/javascript/src/models/cacheCreateNamespaceOut.ts
// this file is @generated
import {
    type EvictionPolicy,
    EvictionPolicySerializer,
} from './evictionPolicy';

export interface CacheCreateNamespaceOut {
    name: string;
    evictionPolicy: EvictionPolicy;
    created: Date;
    updated: Date;
}

export const CacheCreateNamespaceOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): CacheCreateNamespaceOut {
        return {
            name: object['name'],
            evictionPolicy: EvictionPolicySerializer._fromJsonObject(object['eviction_policy']),
            created: new Date(Number(object['created'])),
            updated: new Date(Number(object['updated'])),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: CacheCreateNamespaceOut): any {
        return {
            'name': self.name,
            'eviction_policy': EvictionPolicySerializer._toJsonObject(self.evictionPolicy),
            'created': self.created.getTime(),
            'updated': self.updated.getTime(),
        };
    }
}

File: z-clients/rust/src/models/cache_create_namespace_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::eviction_policy::EvictionPolicy;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct CacheCreateNamespaceOut {
    pub name: String,

    pub eviction_policy: EvictionPolicy,

    pub created: u64,

    pub updated: u64,
}

impl CacheCreateNamespaceOut {
    pub fn new(name: String, eviction_policy: EvictionPolicy, created: u64, updated: u64) -> Self {
        Self {
            name,
            eviction_policy,
            created,
            updated,
        }
    }
}
